src/python/pants/backend/experimental/python/lockfile_test.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from unittest.mock import MagicMock
+
+from pants.backend.experimental.python.lockfile import (
+    PythonLockfileRequest,
+    lockfile_metadata,
+    read_lockfile_metadata,
+    validated_lockfile_content,
+)
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.util.ordered_set import FrozenOrderedSet
+
+
+def test_metadata_round_trip():
+
+    val = "help_im_trapped_inside_a_unit_test_string"
+    output = read_lockfile_metadata(lockfile_metadata(val))
+    assert val == output["invalidation digest"]
+
+
+def test_validated_lockfile_content():
+
+    req = MagicMock(hex_digest="000faaafcacacaca")
+    content = b"""dave==3.1.4 \\
+    --hash=sha256:cab0c0c0c0c0dadacafec0c0c0c0cafedadabeefc0c0c0c0feedbeeffeedbeef \\
+    """
+
+    output = b"""
+# --- BEGIN LOCKFILE METADATA: DO NOT EDIT OR REMOVE ---
+# invalidation digest: 000faaafcacacaca
+# --- END LOCKFILE METADATA ---
+dave==3.1.4 \\
+    --hash=sha256:cab0c0c0c0c0dadacafec0c0c0c0cafedadabeefc0c0c0c0feedbeeffeedbeef \\
+    """
+
+    # Helper function to make the test case more resilient to reformatting
+    line_by_line = lambda b: [ii for i in b.splitlines() if (ii := i.strip())]
+    assert line_by_line(validated_lockfile_content(req, content)) == line_by_line(output)
+
+
+def test_hex_digest_empty():
```

Comment:
There's a lot of duplication in these hash functions. How about using a single test and using `pytest.mark.parametrize` to parametrize the requirements, interpreter_constraints, and expected_hex_digest? https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
good call, I'll make that change